amm/runtime/src/main/scala/ammonite/runtime/ClassLoaders.scala

```diff
@@ -154,4 +157,25 @@ class SpecialClassLoader(parent: ClassLoader, parentSignature: Seq[(Path, Long)]
       case _ => Nil
     })
   }
+
+  override def findResource(name: String) =
+    getURLFromFileDict(name).getOrElse(super.findResource(name))
+
+  override def findResources(name: String) = getURLFromFileDict(name) match {
+    case Some(u) => Collections.enumeration(Collections.singleton(u))
+    case None    => super.findResources(name)
+  }
+
+  private def getURLFromFileDict(name: String) = {
+    val className = name.stripSuffix(".class").replace('/', '.')
+    newFileDict.find(_._1 == className) map { c =>
```

Comment:
Could this be replaced by `newFileDict.get(className)`? That will let it be O(1) rather than O(n) as the size of `newFileDict` grows